pandas/core/generic.py

```diff
@@ -1214,10 +1217,58 @@ def _set_is_copy(self, ref=None, copy=True):
             else:
                 self.is_copy = None
 
+    def _check_copy_on_write(self):
+
+        # we could have a copy-on-write scenario
+        if self.is_copy and self._allow_copy_on_write:
+
+            # we have an exception
+            if isinstance(self.is_copy, Exception):
+                raise self.is_copy
+
+            def get_names_for_obj(__really_unused_name__342424__):
```

Comment:
This probably does not work for 

```
l = [df[...]]
l[0][...] = "x"
```
See here: https://github.com/jreback/pandas/commit/8b684b97523f71d6d0d4da40d8b4edc3e4d820d4#commitcomment-13034569